plugins/languages/typescript-tools.nix

```diff
@@ -0,0 +1,95 @@
+{
+  lib,
+  pkgs,
+  helpers,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.typescript-tools;
+in {
+  extraPlugins = [cfg.package];
+  options.plugins.typescript-tools = {
+    enable = mkEnableOption "typescript-tools";
+    package =
+      helpers.mkPackageOption "typescript-tools"
+      pkgs.vimPlugins.typescript-tools-nvim;
+
+    onAttach =
+      helpers.defaultNullOpts.mkLuaFn "null"
```

Comment:
I _think_ we should set the default value of this to `__lspOnAttach` (so avoiding the use of a defaultNullOpts).
We do some heavy lifting in __lspOnAttach to handle integrations with other plugins, so it'd be nice to benefit from it here too.

Reply:
I'll change it, but I'm not understanding what this does. I know you answered me before, but where can I find where it does?
I know `mkRaw` takes a thing and makes it into raw Lua code. However when I search for `__lspOnAttach` [this line of code](https://github.com/nix-community/nixvim/blob/ddce82d0d1fb88c957ecb7debed2ae6c63b60750/plugins/lsp/default.nix#L152) is as close as I can get. I highly, highly doubt this is it, because it doesn't tell me much.

Reply:
This is exactly that!

If you  search for `onAttach` in nixvim there plugins that add things to it

Reply:
I think I get it now. We're inserting `__lspOnAttach` as raw Lua code, thus it references the value assigned to this in the generated init.lua. The value assigned to `__lspOnAttach` is determined by `plugins.lsp.onAttach`!

Reply:
Yes, we only do that sort of trickery for `__lspOnAttach` I think